crates/p2p_protocol/src/p2p.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{error::P2pResult, p2p_capnp};
+
+/// a message used to verify connectivity with a remote node
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct MsgPing {
+    /// set the milliseconds since unix epoch when sending this message
+    pub ping_send_epoch_ms: u64,
+}
+
+/// a response to a ping message
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub struct MsgPong {
+    /// copy the milliseconds since unix epoch from the triggering ping
+    pub ping_send_epoch_ms: u64,
+    /// set our own machine milliseconds since unix epoch here
+    /// can be used for heuristics about relative clock skew
+    pub ping_received_epoch_ms: u64,
+}
+
+/// an enum representing the various p2p message types that can be sent
+/// between lib3h nodes
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq, Hash)]
+pub enum P2pMessage {
+    MsgPing(MsgPing),
+    MsgPong(MsgPong),
+}
+
+/// get the current system milliseconds since unix epoch
+fn now_ms() -> u64 {
+    let out = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)
+        .expect("duration since unix epoch");
+    out.as_secs() * 1000 + u64::from(out.subsec_nanos()) / 1_000_000
+}
```

Comment:
A bit worried of not using our `since_epoch_ms()` here, i.e. of not using a centralized clock.

Reply:
agree, but, alas, they are in separate crates... Added a note to #211 -

> Also we're using SystemTime in at least two different crates (lib3h + lib3h_p2p_protocol) so we should maybe make the time mocking be a separate crate that others could depend on.